In [13]:
import sys
import os
sys.path.insert(0, '/opt/research/EchoHiding/')
from echohiding import echo_hide, extract_echo_bits, get_mp3_encoded
import pathlib
import librosa
import numpy
from util_functions import *

In [11]:
L = 2048
avgs = []

directory = '/opt/research/fma_medium/'

In [ ]:
# Insert all 0's into the audio clips

y, sr = librosa.load("/opt/research/fma_medium/000/000002.mp3")

bitrate = 64
b_est_mp3 = extract_echo_bits(get_mp3_encoded(y, sr, bitrate), L)

zeros = numpy.ones(len(b_est_mp3))
z = echo_hide(y, L, zeros, alpha=0.2)

b_est_mp3 = extract_echo_bits(get_mp3_encoded(z, sr, bitrate), L)
print(numpy.mean(b_est_mp3))

In [ ]:
# Calc averages of each subdirectory of audio clips and save to a flat file

results = []
# using with statement
with open('natural_avgs.txt', 'a') as file:
    for i in range(156):
        avg = process_dir("{}{}".format(directory, "{0:03}".format(i)), get_avg_echo_vals)
        print("Directory: {} | Average: {}".format("{}{}".format(directory, "{0:03}".format(i)), avg))
        file.write("{}, ".format(avg))
        file.flush()

In [17]:
# Read data back from flat file and process
data = []
with open("natural_avgs.txt", "r") as file:
    txt = file.read()
    txt = txt.strip()
    txt = txt.split(",")
    txt = txt[:-1]

total = 0
ctr = 0

for each in txt:
    total += float(each)
    ctr += 1

avg = total/ctr
print(avg)


0.7374216270845926


The natural average of all audio clips in the data-set combined is: 0.7374216270845926

This implies that the distrobution between 0's and 1's is not 50/50 and rather 27/73.